File: 双指针/283.移动零.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l=0,r=0;
        for(;r<nums.length;r++){
            if(nums[r]!=0){
                int tem = nums[r];
                nums[r]=nums[l];
                nums[l]=tem;
                l++;
            }
        }
    }
}


class Solution {
    public void moveZeroes(int[] nums) {
        int l=0,r=0;
        for(;r<nums.length;r++){
            if(nums[r]!=0){
                nums[l]=nums[r];
                l++;
            }
        }
        for(int i=l;i<nums.length;i++){
            nums[i]=0;
        }
    }
}

File: 双指针/283.移动零.go
package 双指针

func moveZeroes(nums []int) {
	l, r := 0, 0
	for r < len(nums) {
		if nums[r] != 0 {
			tmp := nums[r]
			nums[r] = nums[l]
			nums[l] = tmp
			l++
		}
		r++
	}
}
